include/pistache/client.h

```diff
@@ -136,8 +136,8 @@ struct Connection : public std::enable_shared_from_this<Connection> {
     Private::Parser<Http::Response> parser_;
 };
 
-struct ConnectionPool {
-
+class ConnectionPool {
```

Comment:
Why did you make this change?  What bug/problem does it address?  A class w/ all public methods/member vars is the same as a struct, so this change should be a no-op... but why make it?

Reply:
The reason for this change is that ConnectionPool is forward declared as a class - which gives a warning. Alternatively I can revert this and forward declare it as a struct instead.

Reply:
Ah, I see.  Either direction (all struct or all class) is fine then.